src/libraries/Microsoft.PowerFx.Core/Public/Values/TableValue.cs

```diff
@@ -85,16 +87,40 @@ private static ErrorValue ArgumentOutOfRange(IRContext irContext)
         // Async because derived classes may back this with a network call. 
         public virtual async Task<DValue<RecordValue>> AppendAsync(RecordValue record)
         {
-            // fails by default
-            throw new System.NotImplementedException("It is not possible to append to a TableValue directly.");
+            throw ThrowNotImplementedException(MethodBase.GetCurrentMethod().Name);
+        }
+
+        public virtual async Task<DValue<RecordValue>> PatchCoreAsync(RecordValue originalRecord, RecordValue newRecord)
+        {
+            throw ThrowNotImplementedException(MethodBase.GetCurrentMethod().Name);
+        }
+
+        public async Task<DValue<RecordValue>> PatchAsync(RecordValue originalRecord, RecordValue newRecord)
+        {
+            var recordType = ((TableType)IRContext.ResultType).ToRecord();
```

Comment:
>var recordType = ((TableType)IRContext.ResultType).ToRecord(); [](http://example.com/codeflow?start=12&length=62)

nit:

can we just add a new property:
  public new TableType Type => (TableType)base.Type;

And then this line can be:
  var recordType = Type.ToRecord();